src/model/test/DaylightingDeviceLightWell_GTest.cpp

```diff
@@ -0,0 +1,211 @@
+  Model model;
+
+  Point3dVector points;
+  points.push_back(Point3d(0, 0, 1));
+  points.push_back(Point3d(0, 0, 0));
+  points.push_back(Point3d(0, 1, 0));
+  points.push_back(Point3d(0, 1, 1));
+  SubSurface window(points, model);
+  EXPECT_EQ("FixedWindow", window.subSurfaceType());
+
+  EXPECT_FALSE(window.daylightingDeviceLightWell());
+
+  EXPECT_EQ(0u, model.getConcreteModelObjects<DaylightingDeviceLightWell>().size());
+  DaylightingDeviceLightWell lightWell(window, 1, 2, 3, 0.75);
+  UUID lightWellHandle = lightWell.handle();
+  ASSERT_TRUE(window.daylightingDeviceLightWell());
+  EXPECT_EQ(lightWellHandle, window.daylightingDeviceLightWell()->handle());
+  ASSERT_TRUE(window.addDaylightingDeviceLightWell());
+  EXPECT_EQ(lightWellHandle, window.addDaylightingDeviceLightWell()->handle());
+  EXPECT_EQ(window.addDaylightingDeviceLightWell()->handle(), window.daylightingDeviceLightWell()->handle());
+  EXPECT_EQ(window.handle(), lightWell.subSurface().handle());
+  EXPECT_EQ(1u, model.getConcreteModelObjects<DaylightingDeviceLightWell>().size());
+
+  lightWell.remove();
+  EXPECT_EQ(0u, model.getConcreteModelObjects<DaylightingDeviceLightWell>().size());
+
+  EXPECT_FALSE(window.daylightingDeviceLightWell());
+  ASSERT_TRUE(window.addDaylightingDeviceLightWell());
+  lightWellHandle = window.addDaylightingDeviceLightWell()->handle();
+  EXPECT_EQ(1u, model.getConcreteModelObjects<DaylightingDeviceLightWell>().size());
+  // Calling it a second time while a Light Well already exists returns the existing one, doesn't create a second
+  ASSERT_TRUE(window.addDaylightingDeviceLightWell());
+  EXPECT_EQ(lightWellHandle, window.addDaylightingDeviceLightWell()->handle());
+  EXPECT_EQ(1u, model.getConcreteModelObjects<DaylightingDeviceLightWell>().size());
+
+  ASSERT_TRUE(window.addDaylightingDeviceLightWell());
+  EXPECT_EQ(lightWellHandle, window.addDaylightingDeviceLightWell()->handle());
+  ASSERT_TRUE(window.daylightingDeviceLightWell());
+  EXPECT_EQ(lightWellHandle, window.daylightingDeviceLightWell()->handle());
+
+  // changing to door removes light light well
+  EXPECT_TRUE(window.setSubSurfaceType("Door"));
+  EXPECT_EQ("Door", window.subSurfaceType());
+  EXPECT_FALSE(window.daylightingDeviceLightWell());
+  EXPECT_FALSE(window.addDaylightingDeviceLightWell());
+  EXPECT_EQ(0u, model.getConcreteModelObjects<DaylightingDeviceLightWell>().size());
+
+}
+
+TEST_F(ModelFixture, DaylightingDeviceLightWell_Throw) {
+  Model model;
+
+  Point3dVector points;
+  points.push_back(Point3d(0, 0, 1));
+  points.push_back(Point3d(0, 0, 0));
+  points.push_back(Point3d(0, 1, 0));
+  points.push_back(Point3d(0, 1, 1));
+  SubSurface door(points, model);
+  EXPECT_TRUE(door.setSubSurfaceType("Door"));
+  EXPECT_EQ("Door", door.subSurfaceType());
+  EXPECT_EQ(0, model.getConcreteModelObjects<DaylightingDeviceLightWell>().size());
+
+  EXPECT_ANY_THROW({DaylightingDeviceLightWell lightWell(door);});
```

Comment:
@joseph-robertson note that gtest has macros for that. EXPECT/ASSERT_ANY/NO_THROW are the most common. You could also do `EXPECT_THROW(..., openstudio::Exception)` if you want a specific exception